dashboard/app/models/pd/application/application_base.rb

```diff
@@ -121,27 +76,24 @@ class ApplicationBase < ApplicationRecord
       notes_5
     )
 
+    enum course: %w(
+      csf
+      csd
+      csp
+      csa
+    ).index_by(&:to_sym).freeze
+
     def set_type_and_year
       # Override in derived classes and set to valid values.
       # Setting them to nil here fails those validations and prevents this base class from being saved.
       self.application_year = nil
       self.application_type = nil
     end
 
-    def accepted?
-      status == 'accepted'
-    end
-
-    def unreviewed?
-      status == 'unreviewed'
-    end
-
-    def pending?
-      status == 'pending'
-    end
-
-    def waitlisted?
-      status == 'waitlisted'
+    %w(accepted incomplete pending unreviewed waitlisted).each do |attribute|
+      define_method(:"#{attribute}?") do
+        status == attribute
+      end
```

Comment:
I don't have to do this, but having 20 lines for the same thing was bothering me

Reply:
Yeah, I've run into the problem of not being able to find things in IDEs ... @tim-dot-org would love to hear what you think. Would be happy to revert

Reply:
I could also add a comment that says "generates accepted?, incomplete?, pending?, unreviewed?, waitlisted? methods" but then we get into a potential problem of the comment being out of date

Reply:
yeah, i think the comment would be a good solution! that list has to be manually updated anyways, if the comment is right there it'll be easy to update alongside